src/ptrmask-integration.adoc

```diff
@@ -0,0 +1,16 @@
+[#section_pointer_masking_integration]
+== Integrating {cheri_base_ext_name} and {cheri_default_ext_name} with Pointer Masking
+
+The pointer masking extensions Smmpm, Smnpm, SSnpm, Sspm and Supm are compatible with {cheri_default_ext_name}.
+
+In {cheri_int_mode_name} all data addresses are integer pointers, which
+are interpreted as being XLEN-wide, and may be subject to pointer masking.
+All data accesses are checked against <<ddc>> which is unaffected by
+pointer masking. Therefore no capability bounds encoding is affected.
+
+In {cheri_cap_mode_name} all pointers are capabilities, and the bounds
+encoding relies on XLEN-wide addresses, and therefore the definitions of the bounds
+will need to be updated to be usable for pointer masking.
+
+Therefore supporting pointer masking in {cheri_cap_mode_name} is future work, and so
```

Comment:
I'm not sure this is ideal since it means you also have to mask the PMM bit whenever you do a modesw in hybrid code? I think it would be more consistent to always mask off the bits and just feed those into the bounds check, and rephrase this as "pointer masking is not applied to capability decoding"? 

Setting/clearing the high bits could create an unrepresentable capability (which is why Morello sort of ignored them) and you can't use them for metadata unless you have an almighty capability. As long as the bounds are still decoded relative to the address stored in the capability rather than the masked address, masking can't change the meaning of capabilities. And since it just changes inaccessible virtual addresses I don't think masking prior to the bounds check is a problem? It's not particularly useful but at least it's more consistent than forcing PMM off?

Reply:
ok - I've changed it - please have another look